tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Check: a unit test framework for C
+ * Copyright (C) 2001, 2002 Arien Malec
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+enum {
+    RED,
+    BLUE,
+    PURPLE,
+    YELLOW,
+    BLACK,
+    MAX_TESTS
+} test_ids;
+
+unsigned int test_executed[MAX_TESTS];
+
+static void reset_executed(void)
+{
+    unsigned int i;
+
+    for (i = 0; i < MAX_TESTS; i++)
+    {
+	test_executed[i] = 0;
+    }
+}
+
+START_TEST(red_test1)
+{
+    test_executed[RED] = 1;
```

Comment:
As each of the tests may be run in its own memory space due to using fork(), I'm not sure that this will work as expected. Using the asserts was probably a better way of checking if a specific test was run or not.